### ISSUE TITLE
Problem with macOS Catalina

### ISSUE THREAD

**Issue by sgraillat** [SublimeText/LaTeXTools#1440]
Dear all,

I have updated my OS to macOS Catalina but now I cannot use LaTeXTools anymore. I have re-installed Sublime Text 3 and LaTeXTools but I still have the problem. When I want to compile I have the following errors:

----
TraditionalBuilder: Engine: pdflatex. Invoking latexmk... done.

Could not read log file td3.log

Output from compilation:

Latexmk: This is Latexmk, John Collins, 18 June 2019, version: 4.65.
Latexmk: Changing directory to './'
Rule 'pdflatex': The following rules & subrules became out-of-date:
      'pdflatex'
------------
Run number 1 of rule 'pdflatex'
------------
------------
Running 'pdflatex -synctex=1 -interaction=nonstopmode  -interaction=nonstopmode -synctex=1 -recorder  "td3.tex"'
------------
Latexmk: applying rule 'pdflatex'...
pdflatex: getcwd: Operation not permitted
Latexmk: fls file doesn't appear to have been made.
Failure to make 'td3.pdf'
Collected error summary (may duplicate other messages):
  pdflatex: Command for 'pdflatex' gave return code 1
      Refer to 'td3.log' for details
----------------------
This message may duplicate earlier message.
Latexmk: Failure in processing file 'td3.tex':
   (Pdf)LaTeX didn't generate the expected log file 'td3.log'
----------------------
Latexmk: Undoing directory change
Latexmk: Errors, in force_mode: so I tried finishing targets
---

Help welcome !

Best regards,
Stef

**Comment by sgraillat** [comment]
For other files, I have also this message 👍 
--
TraditionalBuilder: Engine: pdflatex. Invoking latexmk... done.

Could not read log file td3.log

Output from compilation:

pwd: .: Operation not permitted
Latexmk: This is Latexmk, John Collins, 18 June 2019, version: 4.65.
Latexmk: Changing directory to './'
pwd: .: Operation not permitted
pwd: .: Operation not permitted
pwd: .: Operation not permitted
Rule 'pdflatex': The following rules & subrules became out-of-date:
      'pdflatex'
------------
Run number 1 of rule 'pdflatex'
------------
------------
Running 'pdflatex -synctex=1 -interaction=nonstopmode  -interaction=nonstopmode -synctex=1 -recorder  "td3.tex"'
------------
Latexmk: applying rule 'pdflatex'...
shell-init: error retrieving current directory: getcwd: cannot access parent directories: Operation not permitted
pdflatex: getcwd: Operation not permitted
Latexmk: fls file doesn't appear to have been made.
Failure to make 'td3.pdf'
Collected error summary (may duplicate other messages):
  pdflatex: Command for 'pdflatex' gave return code 1
      Refer to 'td3.log' for details
----------------------
This message may duplicate earlier message.
Latexmk: Failure in processing file 'td3.tex':
   (Pdf)LaTeX didn't generate the expected log file 'td3.log'
----------------------
Latexmk: Undoing directory change
Use of chdir('') or chdir(undef) as chdir() is deprecated at /System/Library/Perl/5.18/darwin-thread-multi-2level/Cwd.pm line 507.
Latexmk: Errors, in force_mode: so I tried finishing targets
--

**Comment by r-stein** [comment]
This seems to be related to the underlaying tex distribution. I recommend to open the console ``ctrl+` `` and check for the executed command and then try to reproduce it without LaTeXTools.
Then you may search for the error or ask at texexchange.

An alternative would be to try the basic builder, because it does not use latexmk, but has the processing logic coded internally.

**Comment by sgraillat** [comment]
Thanks for your answer. The problem is not the TeX distribution as it works very well when I use it in Terminal. Here is what I obtain in the console:

---
Welcome to thread Thread-10
Running "pdflatex -interaction=nonstopmode -synctex=1 file.tex"
Finished normally
1
False True True
Expected PDF file /Users/SG/file.pdf not found

**Comment by ig0774** [comment]
Right... the problem is with the distribution version of latexmk (LaTeXTools doesn't bundle this) and probably with latexmk itself. If you change to using the basic builder, things should work.

**Comment by sgraillat** [comment]
No it does not work with the basic builder. Here is what I get:
----
Welcome to thread Thread-7
Running "pdflatex -interaction=nonstopmode -synctex=1 SplitDirectedRoundings.tex"
Finished normally
1
False True True
Expected PDF file /Users/graillat/Documents/Articles/Split/splitrdru/SplitDirectedRoundings.pdf not found

**Comment by ig0774** [comment]
So this is something separate from the latexmk thing. The 1 in the output after the “Finished Normally” message is the return code for pdflatex, which indicates it returned an error. You need to look in the log file for what error occurred (hopefully this error would’ve been displayed in the LaTeXTools output, but that doesn’t display when you have the console opened).

**Comment by ig0774** [comment]
Iirc, `C-l,shift-esc` should bring up the LaTeXTools output panel after you’ve closed the console.

**Comment by sgraillat** [comment]
Thank you for your answer. I did this (C-l,shift-esc) and I obtain this but there is no pdf file generated.

------
[Compiling /Users/SG/file.tex]

Basic Builder: running pdflatex...done.

No errors. No warnings.

/Users/SG/file.log:1: Double-click here to open the full log.

[Done!]

**Comment by ig0774** [comment]
Could you post the contents of the log file?

**Comment by sgraillat** [comment]
Here is the log file:

---
This is pdfTeX, Version 3.14159265-2.6-1.40.20 (TeX Live 2019) (preloaded format=pdflatex 2019.10.24)  26 OCT 2019 18:54
entering extended mode
 restricted \write18 enabled.
 %&-line parsing enabled.
**file.tex
(./file.tex

**Comment by deathaxe** [comment]
Clearly a permission issue. IIRC, GUI apps run sandboxed probably not having same privileges to various resources (filesystem, etc.) as terminal application.

There's not much which can be done to fix it by LaTeXTools.
